app/shared/api/blockchain/transfer/types.ts

```diff
@@ -0,0 +1,22 @@
+import type { KeyringPair } from '@polkadot/keyring/types';
+import { type Hash } from '@polkadot/types/interfaces';
+import type { BN } from '@polkadot/util';
+
+export interface ITransfer {
+  sendTransfer: (params: SendTransferParams) => Promise<Hash>;
+  getTransferFee: (params: FeeParams) => Promise<BN>;
+  getGiftTransferFee: (params: FeeParams) => Promise<BN>;
```

Comment:
I dont think the general transfer interface should know about gifts. Gifts are composed from transfers but not vice versa.
I see what you are trying to achive here, we nee to perform conversion of both transfers for Statemine to calculate USDT fees. 
However, if we think about  independent things that happen here, we will see the following:

1. Any transaction can be paid in sufficient tokens, so we can pay in USDT for anyhting (a separate fee payment service)
2. Transfer interface could eitber
  2.1. Just constuct a tx to pass to fee payment service. This assumes each transfer implementation can only operatie within single tx (should be true for transfers)
  2.2. Calculates the fee itself - more flexibility, more code dublication, worse composbility
3. Gifts that operate with underlying transfers and fees

Overall this is the less sensitive moment rn since I understand this model requires significantly more changes and the current approach doesnt have any critical flaws to address besides potential problems in the future caused by wrong separation of concerns